amr-wind/equation_systems/icns/icns_advection.cpp

```diff
@@ -99,6 +105,26 @@ void MacProjOp::init_projector(const MacProjOp::FaceFabPtrVec& beta) noexcept
 
 void MacProjOp::init_projector(const amrex::Real beta) noexcept
 {
+    m_need_init = false;
+
+    auto& pressure = m_repo.get_field("p");
+    const auto& bctype = pressure.bc_type();
+
+    auto const& lobc = get_projection_bc(
+        amrex::Orientation::low, bctype, m_repo.mesh().Geom());
+    auto const& hibc = get_projection_bc(
+        amrex::Orientation::high, bctype, m_repo.mesh().Geom());
+
+#ifdef AMR_WIND_USE_FFT
+    if (m_use_fft) {
+        if (m_repo.num_active_levels() == 1 && m_has_overset == false) {
+            m_fft_mac_proj = std::make_unique<Hydro::FFTMacProjector>(
+                m_repo.mesh().Geom(0), lobc, hibc);
+            return;
+        }
```

Comment:
Looks like if the user has requested fft, but it is not possible to use, there is no warning or abort to notify the user that it is not being used.

Reply:
Good catch, I can add a warning conditional on the input file having `mac_proj.use_fft = 1`. That would distinguish FFT being automatically enabled/disabled appropriately when AMR-Wind is compiled with FFT vs a user explicitly requesting to use FFT.